### ISSUE TITLE
Memory leak when opening HomePageContextMenu 

### ISSUE THREAD

**Issue by FilippoZazzeroni** [mozilla-mobile/firefox-ios#15495]
## Steps to reproduce

In the home page long press on any tiles, observe the memory allocated by the app increasing even if the context menu is dismissed. 

### Expected behavior

There shouldn't be any memory leak.

### Actual behavior

The memory increases

### Device & build information
* Device: ? 
* Build version: 117.0 (1)
* First seen version: ? 
### Notes
Attachments: 

https://github.com/mozilla-mobile/firefox-ios/assets/55580351/149e43e9-26cc-4ded-b1aa-f5681c4226e1

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-6981)

**Comment by data-sync-user** [comment]
➤ Laurie Marceau commented:

For QA it can be tested the long press menu action sheet (for example on pocket cells in the homepage) has a blur as expected, in both light and dark mode.

**Comment by purifier1990** [comment]
I tried to reproduce this on leak profile, and surprising that it doesn't show the leak signal, but I do observe the memory is increasing.
With that said, I kind of think that the issue might not be a leak but some design code issue that memory leak tool doesn't think this is a leak, but from other aspect, it is. For example, maybe this is controlled by a singleton instance, so the reference is always there and caused memory leak, but that will not be captured by Xcode instrument. That's the only reason I can find to explain what I saw. Will try find some code to dive deep
